spec.html

```diff
@@ -36281,7 +36281,7 @@ <h1>JSON.parse ( _text_ [ , _reviver_ ] )</h1>
         1. Let _jsonString_ be ? ToString(_text_).
         1. Parse UTF16Decode(_jsonString_) as a JSON text as specified in ECMA-404. Throw a *SyntaxError* exception if it is not a valid JSON text as defined in that specification.
         1. Let _scriptString_ be the string-concatenation of `"("`, _jsonString_, and `");"`.
-        1. Let _completion_ be the result of parsing and evaluating _scriptString_ as if it was the source text of an ECMAScript |Script|. The extended PropertyDefinitionEvaluation semantics defined in <emu-xref href="#sec-__proto__-property-names-in-object-initializers"></emu-xref> must not be used during the evaluation.
+        1. Let _completion_ be the result of parsing and evaluating UTF16Decode(_scriptString_) as if it was the source text of an ECMAScript |Script|. The extended PropertyDefinitionEvaluation semantics defined in <emu-xref href="#sec-__proto__-property-names-in-object-initializers"></emu-xref> must not be used during the evaluation.
```

Comment:
is this decoding not implied by "as if it was the source text of an ECMAScript |Script|"?

Reply:
Well, that phrase was introduced in ES 5, when the spec also said:
>  ECMAScript source text is assumed to be a sequence of 16-bit code units for the purposes of this specification.

so (supposedly) there wouldn't have been any "decoding" needed.

It was ES 6 that introduced a clear distinction between Strings and Source Texts, and the need to convert between the two. E.g., that's when the decoding was added to step 2 of this algorithm. I suspect it was just an oversight that it was't added to this step too.

(Nowadays, instead of "parsing and evaluating as if ...", we might involve a call to `ScriptEvaluationJob`.)